scheduler/job.go

```diff
@@ -239,16 +242,32 @@ func (c *collectorJob) Run() {
 		}
 	}
 
+	// Revalidate dependencies in case a new plugin was loaded
+	depGroupMap := gatherMetricsAndPlugins(c.metricManager, c.workflow)
+	for k, val := range depGroupMap {
+		manager, err := c.remoteManagers.Get(k)
+		if err != nil {
+			c.AddErrors(err)
+		}
+
+		errs := manager.ValidateDeps(val.Metrics, val.Plugins)
+		if len(errs) > 0 {
+			for _, err := range errs {
+				c.AddErrors(err)
+			}
+		}
+	}
+
 	metrics := []core.Metric{}
 	for _, rmt := range c.metricTypes {
-		nss, err := c.collector.ExpandWildcards(rmt.Namespace())
+		nss, err := c.metricManager.ExpandWildcards(rmt.Namespace())
 		if err != nil {
 			// use metric directly from the workflow
 			nss = []core.Namespace{rmt.Namespace()}
 		}
 
 		for _, ns := range nss {
-			config := c.configDataTree.Get(ns.Strings())
+			config := c.workflow.configTree.Get(ns.Strings())
```

Comment:
Could You explain me this change? Also, I would like to be sure that this configTree represents both Global and Task Configuration.